Training3/src/task/key-logger/src/component/modal.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {useEffect, useState} from 'react';
+
+type ModalProps = {
+    isOpen: boolean,
+    handleClose: () => void,
+    onKeyPress: (text: string) => void,
+}
+
+const Modal = ({isOpen, handleClose, onKeyPress}: ModalProps) => {
+    const [isRendered, setIsRendered] = useState<boolean>(false)
+
+    let display = isOpen ? 'block' : 'none';
+
+    useEffect(() => {
+        if (isOpen) {
+            setIsRendered(true)
+        }
+    }, [isOpen]);
+
+
+    useEffect(() => {
+        const handleKeyDown = (e: KeyboardEvent) => {
+            onKeyPress(e.key);
+        }
+        const loggerArea = document.getElementById('textArea')
+        if (loggerArea) {
+            loggerArea.addEventListener("keydown", handleKeyDown)
```

Comment:
we need to listen to global documentation not just text-area (user typing without focus on text-area) that's why I suggest use `document.addEventListener`